src/lib/expansion-panel/expansion-panel-adapter.ts

```diff
@@ -70,4 +86,44 @@ export class ExpansionPanelAdapter extends BaseAdapter<IExpansionPanelComponent>
   public animationEnd(): void {
     this._innerElement.style.removeProperty('overflow');
   }
+
+  public tryLocateTargetButton(id: string): void {
+    this._targetButton = this._tryFindTargetButton(id);
+  }
+
+  public initializeAccessibility(): void {
+    const slottedContent = this._defaultSlotElement.assignedElements()[0] as Element | undefined;
+    if (this._targetButton && slottedContent) {
+      const contentElementId = slottedContent.getAttribute('id') ?? `forge-expansion-panel-content-${randomChars()}`;
+      slottedContent.setAttribute('id', contentElementId);
+      this._targetButton.setAttribute('aria-controls', contentElementId);
+    }
+  }
+
+  public updateAriaExpanded(open: boolean): void {
+    if (this._targetButton) {
+      if (open) {
+        this._targetButton.setAttribute('aria-expanded', 'true');
+      } else {
+        this._targetButton.setAttribute('aria-expanded', 'false');
+      }
+    }
```

Comment:
We could use optional chaining here as well, or you could use the `toggleAttribute()` helper from `@tylertech/forge-core` if you prefer. It allows for toggling and setting a value in a one liner.